powerquery-docs/query-folding-basics.md

```diff
@@ -64,20 +58,27 @@ The following diagram explores the process that occurs when a query is evaluated
 5. The results derived from the previous point are loaded to a destination.
 
 >[!NOTE]
->While this example showcases a query with a SQL Database as a data source, the concept applies to queries with or without a data source.
+>While this example showcases a query with an SQL Database as a data source, the concept applies to queries with or without a data source.
+
+## Query folding
+
+When Power Query reads your M script, it runs the script through an optimization process to evaluate your query. This optimization process is called *query folding*. The goal of query folding is to determine which steps (transforms) from your query can be offloaded to your data source that can compute the transformations of your query and which other steps need to be evaluated using the Power Query engine.
 
-When Power Query reads your M script, it runs the script through an optimization process to more efficiently evaluate your query. In this process, it determines which steps (transforms) from your query can be offloaded to your data source. It also determines which other steps need to be evaluated using the Power Query engine.
-This optimization process is called *query folding*, where Power Query tries to push as much of the possible execution to the data source to optimize your query's execution.
+The query folding mechanism in Power Query translates your M script to a language that can be interpreted and executed by your data source. It then pushes the evaluation to your data source and sends the result of that evaluation to Power Query. This operation often provides a much faster query execution than extracting all the required data from your data source and running all transforms required in the Power Query engine.
 
 >[!IMPORTANT]
->All rules from the [Power Query M formula language (also known as the M language)](/powerquery-m/) are followed. Most notably, *lazy evaluation* plays an important role during the optimization process. In this process, Power Query understands what specific transforms from your query need to be evaluated. Power Query also understands what other transforms don't need to be evaluated because they're not needed in the output of your query.
+>All rules from the [Power Query M language](/powerquery-m/) are followed. *lazy evaluation* is important during the optimization process because this is when Power Query understands that only specific transforms from your query need to be evaluated because they are needed in the query output.
```

Comment:
```suggestion
>All rules from the [Power Query M language](/powerquery-m/) are followed. Most notably, *lazy evaluation* is important during the optimization process because this is when Power Query understands that only specific transforms from your query need to be evaluated because they are needed in the query output.
```